crates/topos-config/src/tce.rs

```diff
@@ -6,22 +7,55 @@ use figment::{
     Figment,
 };
 use serde::{Deserialize, Serialize};
+use topos_core::types::ValidatorId;
+use topos_p2p::config::NetworkConfig;
+use topos_tce_transport::ReliableBroadcastParams;
 
-use crate::config::Config;
+use crate::Config;
 use topos_p2p::{Multiaddr, PeerId};
 
 const DEFAULT_IP: std::net::Ipv4Addr = std::net::Ipv4Addr::new(0, 0, 0, 0);
 
-#[derive(Serialize, Deserialize, Debug, Clone)]
+#[derive(Debug)]
+pub enum AuthKey {
+    Seed(Vec<u8>),
+    PrivateKey(Vec<u8>),
```

Comment:
Aren't these known sizes, e.g. 32 bytes? Can we use an array?

Reply:
As above, it is not new code. And no it's not guaranteed that they're 32 bytes if I recall correctly, but I'm not the author of this part, maybe @hadjiszs has more information

Reply:
🤷 Is it important if it's new code or not? I think not.

If we have doubts about the sizes of the seed and private key, perhaps that's an indication that the types used are not expressive enough.

Reply:
Seed can be any size, then `AuthKey::PrivateKey` was with abstracted size to handle different private keys (ed25519 ones of 34bytes, or secp256k1 of 32bytes) as we were using different curve between `PeerId` and `ValidatorId` before

Right now I re-checked and this is used only with secp256k1 private key so always 32 bytes for now

I can create one ticket to refine these types with more explicit types, we could work on it while switching to the edge alternative as these structs are initially fed from the key files generated by edge